test/file.ts

```diff
@@ -2678,6 +2685,82 @@ describe('File', () => {
           });
         });
       });
+
+      it('should recursively create directory and write file contents if destination path is nested', done => {
+        tmp.setGracefulCleanup();
+        tmp.dir(async (err, tmpDirPath) => {
+          assert.ifError(err);
+
+          const fileContents = 'nested-abcdefghijklmnopqrstuvwxyz';
+
+          Object.assign(fileReadStream, {
+            _read(this: Readable) {
+              this.push(fileContents);
+              this.push(null);
+            },
+          });
+
+          const nestedPath = path.join(tmpDirPath, 'a', 'b', 'c', 'file.txt');
+
+          file.download({destination: nestedPath}, (err: Error) => {
+            assert.ifError(err);
+            assert.strictEqual(fs.existsSync(nestedPath), true);
+            fs.readFile(nestedPath, (err, tmpFileContents) => {
+              assert.ifError(err);
+              assert.strictEqual(fileContents, tmpFileContents.toString());
+              done();
+            });
+          });
+        });
+      });
+
+      it('should skip write if asked to write a directory object', done => {
+        // Temporary comment: Option 1: Spy on fs methods
+        const mkdirSync = sandbox.spy(fakeFs, 'mkdirSync');
+        const createWriteStream = sandbox.spy(fakeFs, 'createWriteStream');
+
+        tmp.setGracefulCleanup();
+        tmp.dir(async (err, tmpDirPath) => {
+          assert.ifError(err);
+
+          const fileContents = '';
+
+          Object.assign(fileReadStream, {
+            _read(this: Readable) {
+              this.push(fileContents);
+              this.push(null);
+            },
+          });
+
+          const nestedDir = path.join(tmpDirPath, 'a', 'b', 'c', '/');
+
+          file.download(
+            {destination: nestedDir},
+            (err: Error, buffer: Buffer) => {
+              try {
+                assert.ifError(err);
+
+                // Temporary comment: Option 1: Spy on fs methods
+                assert.strictEqual(createWriteStream.callCount, 0);
+                assert.strictEqual(mkdirSync.callCount, 0);
+
+                // Temporary comment: Option 2: Verify that no folder and/or file were created
+                assert.strictEqual(
+                  fs.existsSync(path.dirname(nestedDir)),
+                  false
+                );
+                assert.strictEqual(fs.existsSync(nestedDir), false);
```

Comment:
Wanted to get your thoughts on what checks make sense here:
- Option 1: Spy on `fs` methods
- Option 2: Check the files and folders written into the temporary directory

There's pros and cons. If this library ever stops using `fs` (unlikely), then the spy wouldn't make sense anymore. On the flip side, checking the folder and files feels a lot like a system-test, since it is reliant on an actual working (linux-based) filesystem.

Reply:
I would lean towards 1 as I don't think we are in any danger of removing `fs` and I don't think these tests should be doing any real disk IO.

Reply:
okay